### PR TITLE
Autoquant

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #38

Summary: Adding autoquantization functionality, using hte do_quant api
we can test kernel speeds and pick the best quantization type (or no
quantization) for each layer.

Test Plan: python test/test.py -k "autoquant"

also tested on SAM and SDXL
https://github.com/pytorch-labs/segment-anything-fast/pull/114
https://github.com/HDCharles/sdxl-fast/commit/8d9942ab05a552f25f5bfe09da02719ce255467f

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D55103983](https://our.internmc.facebook.com/intern/diff/D55103983)

### CODE REVIEW

File: test/test.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
     change_linear_weights_to_int8_woqtensors,
     change_linear_weights_to_int4_woqtensors,
     _replace_with_custom_fn_if_matches_filter,
+    do_autoquant
```
Comment (cpuhrsch) [inline test/test.py:27]: Let's just make this `autoquant`

**Review by cpuhrsch — APPROVED** [review]

**Comment by cpuhrsch** [comment]
Follow up is a refactor and tutorial

**Comment by msaroufim** [comment]
why do we have a new `__init__` file in root folder?

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D55103983).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D55103983).

**Pull request merged**